stochastic_physics/stochastic_physics_wrapper.F90

```diff
@@ -154,6 +170,52 @@ subroutine stochastic_physics_wrapper (GFS_Control, GFS_Data, Atm_block)
             deallocate(skebu_wts)
             deallocate(skebv_wts)
          end if
+         if (GFS_Control%lndp_type .EQ. 2) then ! save wts, and apply lndp scheme 
+             do nb=1,Atm_block%nblks
+                GFS_Data(nb)%Coupling%sfc_wts(:,:) = sfc_wts(nb,1:GFS_Control%blksz(nb),:)
+             end do
+
+             allocate(smc(1:Atm_block%nblks,maxval(GFS_Control%blksz),GFS_Control%lsoil))
+             allocate(slc(1:Atm_block%nblks,maxval(GFS_Control%blksz),GFS_Control%lsoil))
+             allocate(stc(1:Atm_block%nblks,maxval(GFS_Control%blksz),GFS_Control%lsoil))
+             allocate(stype(1:Atm_block%nblks,maxval(GFS_Control%blksz)))
+             allocate(vfrac(1:Atm_block%nblks,maxval(GFS_Control%blksz)))
+             do nb=1,Atm_block%nblks
+                stype(nb,1:GFS_Control%blksz(nb))  = GFS_Data(nb)%Sfcprop%stype(:)
+                smc(nb,1:GFS_Control%blksz(nb),:)  = GFS_Data(nb)%Sfcprop%smc(:,:) 
+                slc(nb,1:GFS_Control%blksz(nb),:)  = GFS_Data(nb)%Sfcprop%slc(:,:) 
+                stc(nb,1:GFS_Control%blksz(nb),:)  = GFS_Data(nb)%Sfcprop%stc(:,:) 
+                vfrac(nb,1:GFS_Control%blksz(nb))  = GFS_Data(nb)%Sfcprop%vfrac(:) 
+             end do
+
+             ! determine whether land paramaters have been over-written 
+             if (mod(GFS_Control%kdt,GFS_Control%nscyc) == 1)  then ! logic copied from GFS_driver
+                    param_update_flag = .true. 
+             else
+                    param_update_flag = .false.
+             endif 
+             call lndp_apply_perts( GFS_Control%blksz, GFS_Control%lsm,  GFS_Control%lsoil, GFS_Control%dtf, & 
```

Comment:
We really need to discuss how this is supposed to work. We don't want any new code in the IPD part of the model (i.e. no new code in `GFS_layer` except variable definitions in `GFS_typedefs.{F90,meta}` and necessary I/O modifications in `GFS_restart.F90`/`GFS_diagnostics.F90`.

Adding file `gfsphysics/GFS_layer/GFS_land_perts.F90` to the IPD and calling it from this wrapper is not the way to go. The wrapper was created to call code from the `stochastic_physics` repository underneath `ufs-weather-model`, not other physics code from CCPP and certainly not from IPD.

Reply:
This looks good to me, thanks for making these changes. Will approve when the regression testing is done and the submodule pointer is reverted back.